testsuite/tests/formatting/margins.ocaml.reference

```diff
@@ -8,6 +8,7 @@ Error: This expression has type
 Line 2, characters 4-9:
   1 + "foo";;
       ^^^^^
-Error: This expression has type string but an expression was expected of type
-         int
+Error: This expression has type 
+       string but an expression was expected of type 
+       int
```

Comment:
I don't know why this is happening. Any ideas?

Reply:
This is yet another memory effect due to the coupling of formatter `margin` and `max_indent` .

More precisely, in this test, the format margin is first set to `20` then to `80` . This has the insidious effect of setting `max_indent` for `Format.std_formatter` to 10.

However, previously, `Location.errorf` printed to a new private formatter which inherited the current margin (and only the margin) from `Format.std_formatter` at each new call of errorf (this setup was done by `Misc.Color.set_color_tag_handling`). Thus, this formatter kept the default `max_indent` value of `68` and since there was no formatting boxes remaining during the printing of `Location.error` to  stdout, the problematic value of  `max_indent` was unseen. 

The fix here is thus to properly set the geometry of `Format.std_formatter` by replacing
`Format.set_margin 80` with `Format.set_margin 80; Format.set_max_indent 70` (in that order!).